reliability: model-switch invalidation fails with EISDIR because faiss.index is a directory, not a file

## What happened

While setting up the server locally against an existing KB whose FAISS index had been built with `sentence-transformers/all-MiniLM-L6-v2`, I switched `OLLAMA_MODEL` to `all-minilm:latest` (new model name). On startup, `FaissIndexManager.initialize` correctly detected the model-name change and tried to invalidate the old index — but crashed:

```
[WARN] Model name has changed from sentence-transformers/all-MiniLM-L6-v2 to all-minilm:latest. Recreating index.
[ERROR] Failed to delete stale FAISS index /home/jean/knowledge_bases/.faiss/faiss.index: Error: EISDIR: illegal operation on a directory, unlink '/home/jean/knowledge_bases/.faiss/faiss.index'
    at async Module.unlink (node:internal/fs/promises:1068:10)
    at async FaissIndexManager.initialize (.../build/FaissIndexManager.js:130:25)
```

## Root cause

`@langchain/community`'s `FaissStore.save(path)` persists to `path/` as a **directory** containing `faiss.index` (binary) + `docstore.json` (metadata):

```
$FAISS_INDEX_PATH/
  faiss.index/          ← directory
    faiss.index         ← binary
    docstore.json       ← metadata
  model_name.txt
```

But `FaissIndexManager.initialize` (around `FaissIndexManager.ts:130`) appears to do `fs.unlink(path.join(FAISS_INDEX_PATH, 'faiss.index'))` — treating that path as a file. On a real on-disk store, the unlink errors with `EISDIR`, the error is logged at `ERROR` level and then re-thrown up through `runStdio` / `run` (also logged), **but the process does not exit** — the server keeps serving, `tools/list` + `list_knowledge_bases` succeed, and the stale (wrong-model) index remains on disk.

Then on the next `retrieve_knowledge` call, the indexer re-embeds changed files with the new model and tries to add them to the pre-existing (different-dim / different-family) FAISS store — so downstream calls may fail with dimension mismatches or just silently produce garbage.

## Why it matters

This is exactly the "model switch" happy path documented in the README and CHANGELOG:
> Model-switch invalidates the index: changing `EMBEDDING_PROVIDER` or the model env var … triggers an auto-rebuild via `FaissIndexManager.initialize()`. Expect a one-time delay on the next `retrieve_knowledge` call while every file is re-embedded.

The docs promise it works. In practice it half-works: the detection fires but the cleanup fails silently and startup proceeds with a poisoned index.

## Suggested fix

In `FaissIndexManager.initialize` (or wherever the stale-index delete happens):

```ts
// Before (approx):
await fs.unlink(path.join(FAISS_INDEX_PATH, 'faiss.index'));
await fs.unlink(path.join(FAISS_INDEX_PATH, 'faiss.index.json')).catch(() => {});

// After:
await fs.rm(path.join(FAISS_INDEX_PATH, 'faiss.index'), { recursive: true, force: true });
```

`fs.rm({ recursive: true, force: true })` handles both cases (file or directory) and is a no-op if the path is missing. Same treatment should probably cover the related \#57 corrupt-index recovery path.

Also: if invalidation fails, startup should probably abort rather than continue with a known-bad index — at minimum, the error shouldn't be swallowed after logging.

## Repro

1. Build a FAISS index with one provider/model (e.g. `EMBEDDING_PROVIDER=huggingface` + default model).
2. Change `EMBEDDING_PROVIDER` or the model env var to anything else.
3. Start the server; observe `EISDIR` in stderr/log.
4. `ls $FAISS_INDEX_PATH/faiss.index/` — still a directory with the old `faiss.index` + `docstore.json` inside.

Surfaced while dogfooding the server as a Claude Code MCP. Reproduced with `@jeanibarz/knowledge-base-mcp-server@0.1.1`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## [Unreleased] — fail stdio startup before exposing poisoned indexes
+
+### Fixed
+
+- **Stdio startup now warms the active FAISS manager before connecting MCP tools.** If index invalidation or corrupt-index cleanup fails, startup aborts before clients can call `tools/list` or `list_knowledge_bases` against a process that already knows its active index is unsafe. Closes #85.
+
 ## [Unreleased] — lazy-import unused embedding providers
 
 ### Changed

--- a/src/KnowledgeBaseServer.test.ts
+++ b/src/KnowledgeBaseServer.test.ts
@@ -440,6 +440,18 @@ describe('KnowledgeBaseServer handlers', () => {
     expect(text).not.toContain(`"source": ${JSON.stringify(betaSource)}`);
   });
 
+  it('runStdio aborts before MCP connect when active model initialization fails (#85)', async () => {
+    await setRetrieveEnv();
+    initializeMock.mockRejectedValueOnce(new Error('stale FAISS cleanup failed'));
+
+    const server = await freshServer();
+    const connectMock = jest.fn().mockResolvedValue(undefined);
+    server['mcp'].connect = connectMock;
+
+    await expect(server['runStdio']()).rejects.toThrow('stale FAISS cleanup failed');
+    expect(connectMock).not.toHaveBeenCalled();
+  });
+
   // --- tool description overrides (#52, RFC 010 M2) -------------------------
   //
   // These tests assert behaviour visible at the MCP wire surface: the

--- a/src/KnowledgeBaseServer.ts
+++ b/src/KnowledgeBaseServer.ts
@@ -300,12 +300,13 @@ export class KnowledgeBaseServer {
   }
 
   private async runStdio(): Promise<void> {
+    // Fail before exposing tools over stdio. If FAISS initialization knows the
+    // active index is bad but cannot clean it up, serving even "safe" tools
+    // leaves clients attached to a poisoned process (#85).
+    await this.warmActiveManager();
     const transport = new StdioServerTransport();
     await this.mcp.connect(transport);
     logger.info('Knowledge Base MCP server running on stdio');
-    // RFC 013: warm up the active model's manager so the first agent call
-    // doesn't pay construction cost.
-    await this.warmActiveManager();
     this.startTriggerWatcher();
   }
 

